### PR TITLE
RIA-5974 - Initial setup for ia-hearings-api

### PR DESCRIPTION
**Pull request by Dbarnes-v1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-5974

### Change description ###
Initial setup for ia-hearings-api

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by thomast1906 — CHANGES_REQUESTED** [review]
As this is a new app, could we move this all to fluxv2 including app deployment?

**Comment by Dbarnes-v1** [comment]
> As this is a new app, could we move this all to fluxv2 including app deployment?

Would this include all other deployments under IA being moved over to v2 as well?

File: k8s/namespaces/ia/ia-hearings-api/ia-hearings-api.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  values:
+    java:
+      disableTraefikTls: true
+      replicas: 2
+      autoscaling:
+        enabled: true
+        maxReplicas: 5
+        targetCPUUtilizationPercentage: 80
+      useInterpodAntiAffinity: true
+      image: hmctspublic.azurecr.io/ia/hearings-api:latest
```
Comment (thomast1906) [inline k8s/namespaces/ia/ia-hearings-api/ia-hearings-api.yaml:20]: Production should be as example  `prod-4f6aa00-20220718112253`
- ImagePolicy also needs added 

Example
`image: hmctspublic.azurecr.io/ia/hearings-api:prod-4f6aa00-20220718112253 #{"$imagepolicy": "flux-system:ia-hearings-api"}`

Comment (Dbarnes-v1) [reply k8s/namespaces/ia/ia-hearings-api/ia-hearings-api.yaml]: Thanks Thomas, added those now.

File: k8s/demo/common-overlay/ia/kustomization.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ patchesStrategicMerge:
 - ../../../namespaces/ia/ia-case-documents-api/demo.yaml
 - ../../../namespaces/ia/ia-case-api/demo.yaml
 - ../../../namespaces/ia/ia-bail-case-api/demo.yaml
+- ../../../namespaces/ia/ia-hearings-api/demo.yaml
```
Comment (thomast1906) [inline k8s/demo/common-overlay/ia/kustomization.yaml:14]: all kustomize need to be added to fluxv2 - as mentioned, lets have a call

File: apps/ia/demo/base/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../base
+patchesStrategicMerge:
+  - ../../ia-hearings-api/demo.yaml
+  - ../../identity/aat.yaml
```
Comment (thomast1906) [inline apps/ia/demo/base/kustomization.yaml:7]: Is demo going to use aat identity? you should have a demo identity

Comment (Dbarnes-v1) [reply apps/ia/demo/base/kustomization.yaml]: Thanks Thomas, added this under the latest commit

**Review by rajeshthuraiyur — DISMISSED** [review]
LGTM

File: apps/ia/ia-hearings-api/demo.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+kind: HelmRelease
+metadata:
+  name: ia-hearings-api
+  namespace: ia
+spec:
+  releaseName: ia-hearings-api
+  values:
+    java:
+      disableTraefikTls: true
+      image: hmctspublic.azurecr.io/ia/hearings-api:prod-8553037-20220727161248
```
Comment (thomast1906) [inline apps/ia/ia-hearings-api/demo.yaml:11]: Missing imagePolicy

File: apps/ia/ia-hearings-api/demo.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      replicas: 2
+      autoscaling:
+        enabled: true
+        maxReplicas: 5
+        targetCPUUtilizationPercentage: 80
```
Comment (thomast1906) [inline apps/ia/ia-hearings-api/demo.yaml:16]: Looks to be alot of duplicate content from base

File: apps/ia/ia-hearings-api/demo.yaml (hunk 4 above)
Comment (Dbarnes-v1) [reply apps/ia/ia-hearings-api/demo.yaml]: thanks, added in latest commit

File: apps/ia/ia-hearings-api/demo.yaml (hunk 5 above)
Comment (Dbarnes-v1) [reply apps/ia/ia-hearings-api/demo.yaml]: thanks, removed these duplicates in latest commit

**Pull request closed without merging**